init/Kconfig

```diff
@@ -2181,3 +2181,20 @@ config ARCH_HAS_SYNC_CORE_BEFORE_USERMODE
 # <asm/syscall_wrapper.h>.
 config ARCH_HAS_SYSCALL_WRAPPER
 	def_bool n
+
+config HAS_RUST
+        def_bool $(success,rustc --version)
+
+config HAS_CARGO
+        def_bool $(success,cargo --version)
+
+config HAS_CARGO_XBUILD
+        def_bool $(success,cargo xbuild --version)
```

Comment:
`cargo xbuild` requires nightly Rust. I would suggest that we get the target upstreamed into the Rust project and into stable Rust, which will make it much more palatable to the Linux kernel.

Reply:
Do you think this is a hard requirement for Rust support moving into mainline? I want to make sure I have my yak stack^W^W dependency chain right :-)

Reply:
While we do have optional things in the kernel that require third-party tools and we also preemptively add support for upcoming GCC (stable) releases, this is about adding support for a new language which will likely require treewide changes to do properly. Therefore, yes, it would help a lot to show that the proposed changes are reasonably stable.

Reply:
Not a *hard* requirement, but you're going to have enough interesting arguments getting this upstream that if we can forestall a couple of them we should. :)

Reply:
Compelling enough for me! Since I know you're also involved in upstream Rust work, do you have an opinion on process for this? Does it need an RFC, or is a PR enough?

Reply:
A PR should suffice. CC me and I'll review it.

Reply:
Awesome -- fine to start with just x86_64?